webhooks-extension/test/util.sh

```diff
@@ -0,0 +1,215 @@
+    ko apply -f config -n $namespace
+    popd
+  fi
+}
+
+# Install https://github.com/tektoncd/experimental/tree/master/webhooks-extension
+function install_webhooks_extension() { 
+  pushd $GOPATH/src/github.com/tektoncd/experimental/webhooks-extension
+  if [ -z "$1" ]; then
+    echo "Usage ERROR for function: install_webhooks_extension [target-namespace]"
+    echo "Missing [namespace]"
+    exit 1
+  fi 
+  namespace=$1
+  docker login
+  dep ensure -v
+  ko apply -f config -n $namespace
+  popd
+  wait_for_ready_pods $namespace 60 10
+}
+
+# Wait until all pods in a namespace are Running or Complete
+# wait_for_ready [namespace] [timeout] <sleepTime>
+# <sleepTime> is optional
+function wait_for_ready_pods() {
+    if [ -z "$1" ] || [ -z "$2" ]; then
+        echo "Usage ERROR for function: wait_for_ready_pods [namespace] [timeout] <sleepTime>"
+        [ -z "$1" ] && echo "Missing [namespace]"
+        [ -z "$2" ] && echo "Missing [timeout]"
+        exit 1
+    fi
+    namespace=$1
+    timeout=$2
+    sleepTime=${3:-"2"}
+
+    # Loop check for ready resources
+    emptyResponse="No resources found."
+    # kubectl command prints to stderr, so redirect it to stdout
+    response=$(kubectl get pods --namespace $namespace --field-selector status.phase!=Running,status.phase!=Succeeded 2>&1)
+    readyResp=$(kubectl get pods --namespace $namespace -o json | jq '.items[]
+        | {phase: .status.phase, conditions: .status.conditions}
+        | select(.phase == "Running")
+        | .conditions[]
+        | select(.type == "Ready")
+        | select(.status != "True")')
+    ctr=0
+    until [ "$response" = "$emptyResponse" ] && [ "$readyResp" = "" ]; do
+        echo "waiting for pods in namespace $namespace:"
+        echo "$response"
+        if [ "$response" = "$emptyResponse" ]; then
+            echo "waiting for ready:"
+            echo "$readyResp" | jq '.'
+        fi
+        if [ "$timeout" -le "$ctr" ]; then
+            echo "ERROR: exceeded timeout (${timeout}s) for namespace '${namespace}'"
+            kubectl get pods --namespace $namespace
+            kubectl describe pods --namespace $namespace
+            return 1
+        fi
+        sleep "$sleepTime"
+        ctr=$((ctr+sleepTime))
+        # kubectl command prints to stderr, so redirect it to stdout
+        response=$(kubectl get pods --namespace $namespace --field-selector status.phase!=Running,status.phase!=Succeeded 2>&1)
+        readyResp=$(kubectl get pods --namespace $namespace -o json | jq '.items[]
+        | {phase: .status.phase, conditions: .status.conditions}
+        | select(.phase == "Running")
+        | .conditions[]
+        | select(.type == "Ready")
+        | select(.status != "True")')
+    done
+    kubectl get pods --namespace $namespace
+}
+
+function check() { 
+  if [ -z "$1" ]; then
+    echo "Usage ERROR for function: check [varname] (example-value)"
+    echo "Missing [varname]"
+    exit 1
+  fi
+  vToCheck=${!1}
```

Comment:
Indirection is nicer than using eval that I have done in the past. I will be using this too.